src/main/java/tech/autodirect/api/database/TableOffers.java

```diff
@@ -0,0 +1,219 @@
+        stmt.setBigDecimal(2, loan_amount);
+        stmt.setBigDecimal(3, capital_sum);
+        stmt.setBigDecimal(4, interest_sum);
+        stmt.setBigDecimal(5, total_sum);
+        stmt.setDouble(6, interest_rate);
+        stmt.setDouble(7, term_mo);
+        stmt.setString(8, installments);
+        stmt.setBoolean(9, claimed);
+
+        // execute and close the above SQL statement
+        stmt.executeUpdate();
+        stmt.close();
+
+        // retrieve the offer ID
+        Statement stmt_lastval = this.db_conn.createStatement();
+        ResultSet rs = stmt_lastval.executeQuery("SELECT LASTVAL();");
+        rs.next();
+        int offer_id = rs.getInt("lastval");
+        stmt.close();
+
+        return offer_id;
+    }
+
+    public void removeOfferByOfferId(int offer_id) throws SQLException {
+        // construct a prepared SQL statement deleting the specified offer
+        PreparedStatement stmt = this.db_conn.prepareStatement(
+        "DELETE FROM " + this.table_name + " WHERE offer_id = ?;"
+        );
+        stmt.setInt(1, offer_id);
+
+        // execute and close the above SQL statement
+        stmt.executeUpdate();
+        stmt.close();
+    }
+
+    public void removeAllOffers() throws SQLException {
+        Statement stmt = this.db_conn.createStatement();
+        stmt.executeUpdate("DELETE FROM " + this.table_name + ";");
+        stmt.close();
+    }
+
+    private static HashMap offerResultToHashMap(ResultSet rs) throws SQLException {
+        return new HashMap<String, Object>() {{
+            put("offer_id", rs.getInt("offer_id"));
+            put("car_id", rs.getInt("car_id"));
+            put("loan_amount", rs.getBigDecimal("loan_amount"));
+            put("capital_sum", rs.getBigDecimal("capital_sum"));
+            put("interest_sum", rs.getBigDecimal("interest_sum"));
+            put("total_sum", rs.getBigDecimal("total_sum"));
+            put("interest_rate", rs.getDouble("interest_rate"));
+            put("term_mo", rs.getDouble("term_mo"));
+            put("installments", rs.getString("installments"));
+            put("claimed", rs.getBoolean("claimed"));
+        }};
+    }
+
+    public HashMap getOfferByOfferId(int offer_id) throws SQLException {
+        // construct a prepared SQL statement selecting the specified offer
+        PreparedStatement stmt = this.db_conn.prepareStatement(
+        "SELECT FROM " + this.table_name + " WHERE offer_id = ?;"
+        );
+        stmt.setInt(1, offer_id);
+
+        // execute the above SQL statement and extract result into a HashMap
+        ResultSet rs = stmt.executeQuery();
+        rs.next();
+        HashMap<String, Object> offer_result = offerResultToHashMap(rs);
+        stmt.close();
+
+        return offer_result;
+    }
+
+    public ArrayList getAllOffers() throws SQLException {
+        // construct a prepared SQL statement selecting all offers
+        Statement stmt = this.db_conn.createStatement();
```

Comment:
This is not a prepared statement, right?

Reply:
Ah cuz here we are not inserting values so there is no injection risk.
The `table_name` cannot be compiled as part of prepared statements afaik.

Reply:
@sammdu Right, I mean that the comment says it's a prepared statement when it's not.

Reply:
> Ah cuz here we are not inserting values so there is no injection risk. The `table_name` cannot be compiled as part of prepared statements afaik.

As a side note, I'm pretty sure you can still use a PreparedStatement by just passing "SELECT * FROM " + this.table_name + ";" to the constructor. Cause it's just a string.